test/integration/integration_test.cc

```diff
@@ -24,21 +28,21 @@ TEST_F(IntegrationTest, Echo) {
   EXPECT_EQ("hello", response);
 }
 
-TEST_F(IntegrationTest, RouterNotFound) { testRouterNotFound(Http::CodecClient::Type::HTTP1); }
+TEST_P(IntegrationTest, RouterNotFound) { testRouterNotFound(Http::CodecClient::Type::HTTP1); }
```

Comment:
@htuch I'll add upstream support in the next PR. Will add a TODO for that.

The current framework probably can't be extended easily to test for arbitrary combinations of IPv4 and Ipv6 address combinations. If that's desired, we can discuss here and I apply the changes to this PR.